src/retroshare/rsposted.h

```diff
@@ -213,16 +286,54 @@ class RsPosted : public RsGxsIfaceHelper, public RsGxsCommentService
 //virtual bool createNewComment(uint32_t &token, RsGxsComment &comment) = 0;
 //virtual bool createNewVote(uint32_t &token, RsGxsVote &vote) = 0;
 
+	/**
+	 * @brief toggle message read status
+	 * @deprecated
+	 * @param[out] token GXS token queue token
+	 * @param[in] msgId
+	 * @param[in] read
+	 */
     RS_DEPRECATED_FOR(setPostReadStatus)
     virtual void setMessageReadStatus(uint32_t& token, const RsGxsGrpMsgIdPair& msgId, bool read) = 0;
         //////////////////////////////////////////////////////////////////////////////
-
-virtual bool createGroup(uint32_t &token, RsPostedGroup &group) = 0;
-virtual bool createPost(uint32_t &token, RsPostedPost &post) = 0;
-
-virtual bool updateGroup(uint32_t &token, RsPostedGroup &group) = 0;
-
-    virtual bool groupShareKeys(const RsGxsGroupId& group,const std::set<RsPeerId>& peers) = 0 ;
+	/**
+	 * @brief Request board creation.
+	 * The action is performed asyncronously, so it could fail in a subsequent
+	 * phase even after returning true.
+	 * @param[out] token Storage for RsTokenService token to track request
+	 * status.
+	 * @param[in] group Board data (name, description...)
+	 * @return false on error, true otherwise
+	 */
+	virtual bool createGroup(uint32_t &token, RsPostedGroup &group) = 0;
+
+	/**
+	 * @brief Create board post. Blocking API.
+	 * @jsonapi{development}
+	 * @param[inout] post
+	 * @return false on error, true otherwise
+	 */
+	virtual bool createPost(uint32_t &token, RsPostedPost &post) = 0;
+
+	/**
+	 * @brief Request board change.
+	 * The action is performed asyncronously, so it could fail in a subsequent
+	 * phase even after returning true.
+	 * @param[out] token Storage for RsTokenService token to track request
+	 * status.
+	 * @param[in] group board data (name, description...) with modifications
+	 * @return false on error, true otherwise
+	 */
+	virtual bool updateGroup(uint32_t &token, RsPostedGroup &group) = 0;
+
+	/**
+	 * @brief Share board publishing key
+	 * This can be used to authorize other peers to post on the board
+	 * @param[in] group Board id
+	 * @param[in] peers peers to which share the key
+	 * @return false on error, true otherwise
+	 */
+	virtual bool groupShareKeys(const RsGxsGroupId& group,const std::set<RsPeerId>& peers) = 0 ;

```

Comment:
These look like gxs related function (usage of token). Is it necessary to expose these?
They are also redundant to other functions above (e.g. `virtual bool createBoard(RsPostedGroup& board) =0;` vs `virtual bool createGroup(uint32_t &token, RsPostedGroup &group) = 0;`

Reply:
groupsharekeys not needed on boards normaly i wasnt sure if needed these.

updategroup is now editboard right 	
`virtual bool editBoard(RsPostedGroup& board) =0;
`
but createpost is not available with api

Reply:
ok, then please only expose `createPost`

Reply:
i has updated look